server/graphql-schema.js

```diff
@@ -113,54 +116,71 @@ const graphqlSchema = gql`
         testPlanReports(status: TestPlanReportStatus): [TestPlanReport]!
     }
 
-    interface Test {
+    interface BaseTest {
         title: String!
         index: Int!
+        referenceFilePath: String!
+    }
+
+    type Test implements BaseTest {
+        title: String!
+        index: Int!
+        referenceFilePath: String!
         # TODO: account for running scripts
         instructions: [Instruction]!
         assertions: [Assertion]!
+        passThroughs: [PassThrough]!
         assertionCount: Int!
         optionalAssertionCount: Int!
-        passThroughs: [PassThrough]!
     }
 
     type Instruction {
         # TODO: account for automation
         manualInstruction: String!
     }
 
-    interface Assertion {
+    interface BaseAssertion {
         # TODO: account for at-specific assertions
         # TODO: account for optional assertions
         # TODO: account for automation
         manualAssertion: String!
     }
 
+    type Assertion implements BaseAssertion {
+        manualAssertion: String!
+        command: String!
+        priority: String!
```

Comment:
Instead of priority I had it in my head that we might be able to just have optional and required assertions. If we're using priority I would definitely want to remove the term optional / required from GraphQL since that would mean we have two terms for the same thing.

Reply:
Well, it is referred to as an assertion priority:
> Assertion priority: Specifies whether an assertion is “must have” or “optional”.

[From _Working Mode_](https://github.com/w3c/aria-at/wiki/Working-Mode)

> just have optional and required assertions

Maintaining that here to me seems to be something like 2 separate booleans that would say `isRequired` or `isOptional`? But the value of `priority` here could realistically just be `required` or `optional` or any other priority 'state' that may be introduced.

> If we're using priority I would definitely want to remove the term optional / required from GraphQL since that would mean we have two terms for the same thing.

I'm not exactly seeing the need for anything to get replaced or removed here

Reply:
> I'm not exactly seeing the need for anything to get replaced or removed here

I was confused about why you were confused so I looked at the schema and saw that while I had included the term `optional`, I never included the term `required`.

That makes what I said ... confusing.

So the previous schema didn't quite model assertion priority well enough to avoid being confusing, and I want to fix that. One way would be to rename the "assertions" and "assertionCount" fields to "requiredAssertions" and "requiredAssertionCount" ... but I think you're actually onto something with the priority field, which is honestly a lot simpler. What about this:

```gql
    type TestResult implements BaseTest {
        title: String!
        index: Int!
        # ...
        assertions(priority: AssertionPriority): [Assertion]!
        assertionCount(priority: AssertionPriority): Int!
        passThroughs: [PassThrough]!
        # optionalAssertionCount: Int! # remove
        # ...
        passThroughResults: [PassThroughResult]!
        assertionsPassed(priority: AssertionPriority): Int!
        # optionalAssertionsPassed: Int! # remove
        unexpectedBehaviorCount: Int!
        # ...
    }
```

Plus an enum for AssertionPriority:

```gql
    enum AssertionPriority {
        REQUIRED
        OPTIONAL
    }
```

This would allow us to keep the priority field, reduce the number of fields, and would allow splitting up the optional and required assertions as needed in the UI. What do you think about that? And do you understand why I think there are two conflicting approaches in place right now?

Reply:
Yep, I understand the disconnect here. And this approach could work